pylint/testutils/output_line.py

```diff
@@ -77,41 +82,117 @@ class OutputLine(NamedTuple):
     symbol: str
     lineno: int
     column: int
+    end_lineno: Optional[int]
+    end_column: Optional[int]
     object: str
     msg: str
     confidence: str
 
     @classmethod
     def from_msg(cls, msg: Message) -> "OutputLine":
+        """Create an OutputLine from a Pylint Message"""
         column = cls._get_column(msg.column)
+        end_line = cls._get_py38_none_value(msg.end_line)
+        end_column = cls._get_py38_none_value(msg.end_column)
         return cls(
             msg.symbol,
             msg.line,
             column,
+            end_line,
+            end_column,
             msg.obj or "",
             msg.msg.replace("\r\n", "\n"),
-            msg.confidence.name if msg.confidence != UNDEFINED else HIGH.name,
+            msg.confidence.name,
         )
 
     @staticmethod
     def _get_column(column: str) -> int:
+        """Handle column numbers with the exception of pylint < 3.8 not having them
+        in the ast parser.
+        """
         if not PY38_PLUS:
             # We check the column only for the new better ast parser introduced in python 3.8
             return 0  # pragma: no cover
         return int(column)
 
+    @staticmethod
+    def _get_py38_none_value(value: T) -> Optional[T]:
+        """Handle attributes that are always None on pylint < 3.8 similar to _get_column."""
+        if not PY38_PLUS:
+            # We check the value only for the new better ast parser introduced in python 3.8
+            return None  # pragma: no cover
+        return value
```

Comment:
That will likely not work for nodes that added `end_lineno` and `end_col_offset` later. E.g. `keyword`, `Slice` (both only in `3.9`). Maybe only compare the values if `end_lineno is not None` or better a new field in `[testoptions]` to overwrite `PY38_PLUS`. I.e. if field is specified, ignore `end_lineno` check for all versions prior.

Reply:
Do we ever create messages on those nodes though? Even `invalid-slice-index` does not pass an `nodes.Slice` as its `node` parameter. Could it be that we're fine if we ignore these?

I'm not sure how the `testoptions` stuff work so I'll need some guidance if want to add a new one.

Reply:
I think ``only compare the values if end_lineno is not None``  is the simpler because I don't know what we could put in the test_option. A condition that needs to succeed or we return None ? Seems like we'll have to launch an eval for that (with all the implication for job launched from fork via github actions) ?

Reply:
I looked into it a bit more and I think `min_pyver_end_line` could work? If we convert it to a boolean and pass it to `_get_py38_none_value` that should make it future proof while testing on versions that it should.

`only compare the values if end_lineno is not None` is what was previously done for `confidence` with the `__eq__`, I don't think that is the way to go as it allows to just never check it.

Reply:
> min_pyver_end_line

Ho, of course. I did not think of that. Very nice solution ! Should we add one option removing the check for line/end_line/column/end_column (``min_pyver_position`` to check all 4 values properly only in the version that is the most accurate) or should we add 4 options (``min_pyver_end_line``, `` min_pyver_end_column`` ... in order to check line/column if line_end or column_end do not exists in other version) ?

Reply:
@cdce8p Knows more about this than I do so I'll let him respond to this but I believe `3.6` has `line` and `col_offset` for all nodes that should have it. `3.8` adds `end_line` and `end_col_offset` for most and `3.9` adds it for some that were missing.

Thus I think we can create one `min_pyver_end_position` (?) which defaults to `3.8.0` and can be increased if needed.

Reply:
Don't worry 😄 ``min_pyver_end_position`` convinced me, do you agree @cdce8p ? By the way it's probably not strictly necessary to release 2.12.0 as we don't have trouble with it right now. We're probably  never emitting message for Slice/Keyword nodes ?

Reply:
Found some time, see #5386